app/portainer/hostmanagement/fdo/fdo.service.ts

```diff
@@ -23,13 +23,66 @@ export async function configureDevice(
   }
 }
 
+export async function createProfile(
+  profileName: string,
+  method: string,
+  profileFileContent: string
+) {
+  const payload: Record<string, unknown> = {
+    Name: profileName,
+    ProfileFileContent: profileFileContent,
+  };
+  try {
+    await axios.post(`${BASE_URL}/profiles`, payload, {
+      params: { method },
+    });
+  } catch (e) {
+    throw parseAxiosError(e as Error, 'Unable to create profile');
+  }
+}
+
 export async function getProfiles() {
   try {
-    const { data: profiles } = await axios.get<Profiles>(
+    const { data: profiles } = await axios.get<Profile[]>(
       `${BASE_URL}/profiles`
     );
-    return profiles.profiles;
+    return profiles;
   } catch (e) {
     throw parseAxiosError(e as Error, 'Unable to retrieve the profiles');
   }
 }
+
+export async function getProfile(profileId: number) {
+  try {
+    const { data: profile } = await axios.get<Profile>(
+      `${BASE_URL}/profiles/${profileId}`
+    );
+    return profile;
+  } catch (e) {
+    throw parseAxiosError(e as Error, 'Unable to retrieve profile');
+  }
+}
+
+export async function deleteProfile(profileId: number) {
+  try {
+    await axios.delete(`${BASE_URL}/profiles/${profileId}`);
+  } catch (e) {
+    throw parseAxiosError(e as Error, 'Unable to delete profile');
+  }
+}
+
+export async function updateProfile(
+  profileId: number,
+  profileName: string,
+  profileFileContent: string
+) {
+  const payload: Record<string, unknown> = {
```

Comment:
same, you don't need to specify type and can use regular camelCase

Reply:
done